lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
Hmm, not sure I agree. Returning false is akin to saying "the predicate tested is valid but does not match". Whereas doing with_captures on a string/string assertion isn't even a valid predicate.

Put another way, `match` returning false with a message is generally intended to be fixed by changing the code under test. However, in this case, the test itself is flawed so I would expect an exception to be raised to indicate the test itself is invalid.